client/state-db/src/pruning.rs

```diff
@@ -306,6 +320,18 @@ impl<BlockHash: Hash, Key: Hash, D: MetaDb> RefWindow<BlockHash, Key, D> {
 			};
 
 		let queue = if count_insertions {
+			// Highly scientific crafted number for deciding when to print the warning!
+			if window_size > 1000 {
+				log::warn!(
+					target: LOG_TARGET,
+					"Detected large pruning window of {window_size} blocks while running with a database \n
+					that isn't supporting ref counting, e.g. RocksDb. \n\
+					This means that all the blocks in the pruning window are stored in memory which \n\
+					may leads to a high memory usage and to possible out of memory crashes. \n\
+					Use for example ParityDb to not store the entire pruning window in memory.\
+				");
```

Comment:
Hmm.... as a *user visible* comment this is perhaps a little to detailed? (: I don't know. Maybe making this a code comment and using something terser and more shouty for the message would be better? Something like:

`Large pruning window of {window_size} detected! THIS CAN LEAD TO HIGH MEMORY USAGE AND CRASHES. Reduce the pruning window or switch your database to paritydb.`

But I don't *really* care either way, so it's up to you.

Reply:
Yours sound better :P Mine was just me trying to bring over the same point, but way more complicated :P